src/audio/module_adapter/module/cadence.c

```diff
@@ -632,11 +632,15 @@ cadence_codec_process(struct processing_module *mod,
 	}
 
 	API_CALL(cd, XA_API_CMD_EXECUTE, XA_CMD_TYPE_DO_EXECUTE, NULL, ret);
-	if (ret != LIB_NO_ERROR) {
+	//if (ret != LIB_NO_ERROR) {
```

Comment:
You might add a define and use it where necessary, like:
```
#define XA_CHECK_ERROR(dev, err)
({
    if (err != LIB_NO_ERROR)
    {
        if (XA_ERROR_SEVERITY(err))
        {
            comp_err(dev, "error: %x), err);
            return err;
        }
        comp_warn(dev, "warning: %x), err);
    }
    err;
})
```

Reply:
I do like this idea, thanks. Will do in the next submission.